.github/workflows/integration-tests.yml

```diff
@@ -37,6 +37,15 @@ jobs:
             echo '::set-output name=matrix-HIP::["ubuntu-latest"]'
           fi
 
+      - name: Enable cache for llvm dependency
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.triton/llvm
+          key: llvm-6f44bb77-ubuntu-x64
+          restore-keys: |
+            llvm-6f44bb77-ubuntu-x64
```

Comment:
This is good for a test, but in practice this hash changes over time.

Also we probably want to cache all of the directories under ~/.triton (other than ~/.triton/cache, if it exists).  But doing just one is fine for a test!

Reply:
+1. I think we can also cache the `pybind11` and `nvidia` directory in `~/.trition`. We need to parameterize the cache key by considering their versions, following https://github.com/actions/cache?tab=readme-ov-file#creating-a-cache-key.

Reply:
Okay I'm trying it out with https://github.com/openai/triton/pull/3562.